crates/ruma-client-api/src/rendezvous/create_rendezvous_session.rs

```diff
@@ -0,0 +1,72 @@
+//! `POST /_matrix/client/*/rendezvous/`
+//!
+//! Create a rendezvous session.
+
+pub mod unstable {
+    //! `msc4108` ([MSC])
+    //!
+    //! [MSC]: https://github.com/matrix-org/matrix-spec-proposals/pull/4108
+
+    use http::header::{CONTENT_TYPE, ETAG, EXPIRES, LAST_MODIFIED};
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata,
+    };
+    use url::Url;
+
+    const METADATA: Metadata = metadata! {
+        method: POST,
+        rate_limited: true,
+        authentication: None,
+        history: {
+            unstable => "/_matrix/client/unstable/org.matrix.msc4108/rendezvous",
+        }
+    };
+
+    /// Request type for the `POST` `rendezvous` endpoint.
+    #[request]
+    pub struct Request {
+        /// Any data up to maximum size allowed by the server.
+        #[ruma_api(body)]
+        pub content: String,
+
+        /// The content type, must be of the `text/plain` type.
+        #[ruma_api(header = CONTENT_TYPE)]
+        pub content_type: String,
+    }
+
+    impl Request {
+        /// Creates a new `Request` with the given content.
+        pub fn new(content: String) -> Self {
+            Self { content, content_type: "text/plain".to_owned() }
+        }
+    }
+
+    impl Default for Request {
+        fn default() -> Self {
+            Self::new("".to_owned())
+        }
+    }
+
+    /// Response type for the `POST` `rendezvous` endpoint.
+    #[response]
+    pub struct Response {
```

Comment:
> It seems like the response can be either this or a 307 redirect. We should support both. This requires manual implementation though, like in https://github.com/ruma/ruma/pull/1797.

Are we sure we want to handle this at the Ruma level? The redirect types wouldn't have anything Matrix specific and they would be just a less generic copy of the standard HTTP resposne type. 

I imagine most HTTP libraries will transparently follow redirects, and only give you the final response. For example [`reqwests` does this in its default configuration](https://docs.rs/reqwest/latest/reqwest/redirect/index.html).

Reply:
I am also thinking about server implementations here.

Reply:
Ok, but I don't see how that changes things. I would expect a server library proxy would have a generic redirection mechanism. Indeed, [`axum` does have exactly that](https://docs.rs/axum/latest/axum/response/struct.Redirect.html), in other cases I might expect a reverse proxy to perform the redirect.

I don't see where Ruma would come into use here. Do you have an example where this would be useful?

Reply:
I know that in [Conduit's SSO login implementation](https://gitlab.com/famedly/conduit/-/merge_requests/587), they use Ruma's response type for the redirect. It's not the only option they would have and I have no idea if they will need it in this case. If you don't want to make the response an enum, it's fine for now.

Reply:
Hmm, that's a bit strange to me. Let's leave the redirect out for now, if Conduit deems it necessary we can revisit this.